src/overmap_ui.cpp

```diff
@@ -1429,12 +1429,21 @@ static tripoint display( const tripoint &orig, const draw_data_t &data = draw_da
                 curs.y = p.y;
             }
         } else if( action == "CHOOSE_DESTINATION" ) {
+            bool in_road_vehicle = g->u.in_vehicle && g->u.controlling_vehicle;
+            const optional_vpart_position vp = g->m.veh_at( g->u.pos() );
+            bool in_boat = false;
+            if( vp && in_road_vehicle ) {
+                vehicle &veh = vp->vehicle();
+                in_boat = veh.can_float() && veh.is_watercraft() && veh.is_watercraft();
```

Comment:
Why Check `veh.is_watercraft()` twice? also does that exclude amphibious vehicle?

Reply:
Meant to be is_in_water()

it dosnt exclude amphibious vehicle, but the autodrive wont allow paths that go over land + water. because land vehicles currently need roads only.